src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityDiode.java

```diff
@@ -35,16 +38,19 @@
 
 import static gregtech.api.capability.GregtechDataCodes.AMP_INDEX;
 
-public class MetaTileEntityDiode extends MetaTileEntityMultiblockPart implements IPassthroughHatch, IMultiblockAbilityPart<IPassthroughHatch> {
+public class MetaTileEntityDiode extends MetaTileEntityMultiblockPart implements IPassthroughHatch, IMultiblockAbilityPart<IPassthroughHatch>, IControllable {
 
     protected IEnergyContainer energyContainer;
 
     private static final String AMP_NBT_KEY = "amp_mode";
+    private static final String WORKING_ALLOWED_NBT_KEY = "WorkingAllowed";
     private int amps;
+    private boolean isWorkingAllowed;
```

Comment:
If the field is changed to indicate the diode is being disabled instead of enabled, you can eliminate the `isWorkingAllowed = true` on ctor and simplify the NBT deserialization a bit, like this:
```java
this.isWorkingDisabled = data.getBoolean(WORKING_DISABLED_NBT_KEY);
```

Which, in my opinion, is a cleaner approach.

Reply:
What will happen to existing diode then?
Wouldn't it then default to false and disable all diodes?

Reply:
If you flip the meaning of the flag around, to let's say something like `isWorkingDisabled`, then preexisting diodes will still work, since it will be set to enabled (`isWorkingDisabled == false`) on deserialization.